### PR TITLE
test_ncpflash: convert to boost::test

### PR DESCRIPTION
**Pull request by akva2**

### CODE REVIEW

**Comment by akva2** [comment]
jenkins build this please

**Comment by akva2** [comment]
jenkins build this please

File: tests/material/test_ncpflash.cpp [hunk 1]
```diff
@@ -177,102 +171,154 @@ inline void testAll()
...
+        // set the fluid temperatures
+        fsRef.setTemperature(T);
+    }
+
+    static Fixture& getInstance()
+    {
+        static std::unique_ptr<Fixture> instance;
+        if (!instance) {
+            instance = std::make_unique<Fixture>();
+        }
```
Comment (bska) [inline tests/material/test_ncpflash.cpp:212]: Why do we use a pointer here?

**Review by bska — COMMENTED** [review]
This looks good for the most part.  The only bit I don't understand is why there's a `unique_ptr<>`.  Would you mind elaborating?

File: tests/material/test_ncpflash.cpp (hunk 1 above)
Comment (akva2) [reply tests/material/test_ncpflash.cpp:212]: probably not a real need, but i wanted to put it on the heap, not the bss.

Comment (akva2) [reply tests/material/test_ncpflash.cpp:213]: sure.

**Comment by akva2** [comment]
jenkins build this please

**Review by bska — APPROVED** [review]
Thanks a lot for the update.  I'll merge into master.

**Pull request merged**